compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -114,35 +124,32 @@ private void generateObjectGraph(TypeElement element) {
           break;
         case SIMPLE_MEMBERS_INJECTION:
         case MEMBERS_INJECTOR:
-          membersInjectionMethods.add(methodDescriptor);
+          memberInjectorsTypeGraph.addVertex(methodDescriptor);
+          memberInjectionMethodsTypeDescriptorMap.put(methodDescriptor.type(), methodDescriptor);
+
+          for (ComponentMethodDescriptor pendingChild : potentialPendingEdges.get(methodDescriptor.type())) {
+            memberInjectorsTypeGraph.addEdge(pendingChild, methodDescriptor);
+          }
+
+          DeclaredType declaredType = (DeclaredType) methodDescriptor.type();
+          TypeElement typeElement = (TypeElement) declaredType.asElement();
+
+          List<TypeMirror> parents = new ArrayList<>(typeElement.getInterfaces());
+          parents.add(typeElement.getSuperclass());
+          for (TypeMirror parent : parents) {
+            ComponentMethodDescriptor parentDescriptor = memberInjectionMethodsTypeDescriptorMap.get(parent);
+            if (parentDescriptor == null) {
+              potentialPendingEdges.put(parent, methodDescriptor);
```

Comment:
Unnecessary if conditions in the cascade will have a direct impact (though possibly negligible) at runtime. I think you'd better pay a bit more at compile-time to save perf at runtime.